release notes/v0.44.0.md

```diff
@@ -0,0 +1,221 @@
+- [#2999](https://github.com/grafana/k6/pull/2999) Directly embed `lib.TestPreInitState` in `js/common.InitEnvironment`.
+- [#2892](https://github.com/grafana/k6/pull/2892) Added z/OS build flags for IBM z/OS compatibility. Thanks, @msradam!
+- [#2833](https://github.com/grafana/k6/pull/2833) Fixed detected memory leaks.
+- [#2931](https://github.com/grafana/k6/pull/2931), [#2940](https://github.com/grafana/k6/pull/2940), [#2895](https://github.com/grafana/k6/pull/2895), [#3002](https://github.com/grafana/k6/pull/3002) Updated k6's dependencies. Added `goja`'s generator support.
+- [#2947](https://github.com/grafana/k6/pull/2947), [#2943](https://github.com/grafana/k6/pull/2943), [#2946](https://github.com/grafana/k6/pull/2946), [#3009](https://github.com/grafana/k6/pull/3009), [#3012](https://github.com/grafana/k6/pull/3012), [#2894](https://github.com/grafana/k6/pull/2894) Tests' fixes, refactoring, and improvements.
+- [#2891](https://github.com/grafana/k6/pull/2891), [#2921](https://github.com/grafana/k6/pull/2921), [#2923](https://github.com/grafana/k6/pull/2923), [#2990](https://github.com/grafana/k6/pull/2990), [#2995](https://github.com/grafana/k6/pull/2995), [#3016](https://github.com/grafana/k6/pull/3016), [#2989](https://github.com/grafana/k6/pull/2989) Linters and formatting fixes.
+- [#3005](https://github.com/grafana/k6/pull/3005) The samples catalog was renamed to examples.
+- [browser#776](https://github.com/grafana/xk6-browser/pull/776) Fixed a test for preset flags for chrome on darwin based systems.
+- [browser#782](https://github.com/grafana/xk6-browser/pull/782), [browser#783](https://github.com/grafana/xk6-browser/pull/783), [browser#826](https://github.com/grafana/xk6-browser/pull/826) Fixed and refactored the mapping.
+- [browser#797](https://github.com/grafana/xk6-browser/pull/797), [browser#832](https://github.com/grafana/xk6-browser/pull/832) Fixed multi browser close.
+- [browser#796](https://github.com/grafana/xk6-browser/pull/796), [browser#810](https://github.com/grafana/xk6-browser/pull/810) Refactored `browserContext.SetExtraHTTPHeaders` to work with errors and ErrFatal.
+- [browser#798](https://github.com/grafana/xk6-browser/pull/798), [browser#799](https://github.com/grafana/xk6-browser/pull/799) Added more tests for the mapping layer.
+- [browser#802](https://github.com/grafana/xk6-browser/pull/802) Added a helper to assert on the logs and dump the logs for easier debugging.
+- [browser#807](https://github.com/grafana/xk6-browser/pull/807) Fixed incorrect keyboard key code on up/down key presses.
+- [browser#819](https://github.com/grafana/xk6-browser/pull/819) Browser.Launch now transitions to Browser.Connect when a CDP url is provided in an env var.
+- [browser#821](https://github.com/grafana/xk6-browser/pull/821), [browser#824](https://github.com/grafana/xk6-browser/pull/824), [browser#830](https://github.com/grafana/xk6-browser/pull/830) Upgraded dependencies and fixed breaking changes.
```

Comment:
Should we have an Announcements section in the release notes? We sometimes mention topics that were not strictly related to code changes, but are important for the direction of the project. This can likely go at the end, or maybe it would better to place it higher, to ensure people actually read it?

For example, it might be worth mentioning our plans with the new HTTP API, and link to the [design document](https://github.com/grafana/k6/blob/master/docs/design/018-new-http-api.md).

Let me know if you agree, and I can come up with a short blurb.

Reply:
> Release notes should not serve as milestones.

Agreed. OK then, we can always make announcements on the forum, or somewhere more appropriate.